### PR TITLE
DOC clean up doc/source/getting_started/overview.rst

### LINKED ISSUE #35980: DOC: doc/source/getting_started/overview.rst
#### Location of the documentation

"https://pandas.pydata.org/docs/getting_started/overview.html"

#### Documentation problem

Unnecessary words and hyphens were missing.

#### Suggested fix for documentation

E.g. real world can be replaced by real-world

### PR DESCRIPTION
**Pull request by 0xpranjal**
- [x] closes #35980

### CODE REVIEW

File: doc/source/getting_started/overview.rst [hunk 1]
```diff
@@ -174,4 +174,3 @@ License
 -------
 
 .. literalinclude:: ../../../LICENSE
-
```
Comment (MarcoGorelli) [inline doc/source/getting_started/overview.rst:177]: I think the blank line at the end of the file was fine, could you revert this?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @Bhard27 !

File: doc/source/getting_started/overview.rst [hunk 2]
```diff
@@ -72,7 +72,7 @@ Some other notes
    extensively tweaked in `Cython <https://cython.org>`__ code. However, as with
    anything else generalization usually sacrifices performance. So if you focus
    on one feature for your application you may be able to create a faster
-   specialized tool.
+  -specialized tool.
```
Comment (MarcoGorelli) [inline doc/source/getting_started/overview.rst:75]: I think this was fine as it was before too, and the hyphen seems out-of-line with the others. Did you try building this page of the docs locally?

File: doc/source/getting_started/overview.rst (hunk 1 above)
Comment (Bhard27) [reply doc/source/getting_started/overview.rst]: I have added the space back in the latest commit.

File: doc/source/getting_started/overview.rst (hunk 2 above)
Comment (Bhard27) [reply doc/source/getting_started/overview.rst:75]: Yeah, I think that is causing a problem in the build. I'll revert it back.

**Review by MarcoGorelli — APPROVED** [review]
LGTM pending green

**Comment by Bhard27** [comment]
All checks have passed without any conflicts. I think we're good to push the code and close the issue.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Bhard27

**Pull request merged**